src/lambda/wizards/samDeployWizard.ts

```diff
@@ -195,27 +197,35 @@ export class SamDeployWizard extends MultiStepWizard<SamDeployWizardResponse> {
     private readonly response: Partial<SamDeployWizardResponse> = {}
 
     public constructor(
+        private readonly contextCommands: Pick<AWSContextCommands, 'onCommandSelectRegion'>,
         private readonly context: SamDeployWizardContext = new DefaultSamDeployWizardContext()
     ) {
         super()
     }
 
     protected get startStep() {
-        return this.TEMPLATE
+        return this.REGION
     }
 
     protected getResult(): SamDeployWizardResponse | undefined {
-        if (!this.response.template || !this.response.s3Bucket || !this.response.stackName) {
+        if (!this.response.region || !this.response.template || !this.response.s3Bucket || !this.response.stackName) {
             return undefined
         }
 
         return {
+            region: this.response.region,
             template: this.response.template,
             s3Bucket: this.response.s3Bucket,
             stackName: this.response.stackName
         }
     }
 
+    private readonly REGION: WizardStep = async () => {
+        this.response.region = await this.contextCommands.onCommandSelectRegion()
+
+        return this.response.region ? this.TEMPLATE : undefined
+    }
+
     private readonly TEMPLATE: WizardStep = async () => {
```

Comment:
`TEMPLATE` needs to go "back" to `REGION`
It also needs a back button now.

Reply:
Implemented (the first revision was before merging back button changes)